lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -1884,6 +1906,46 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     visitPart(RST->getReferentType(), optionalKind);
   }
   
+  Optional<ValueDecl *> findRenamedDecl(const swift::AvailableAttr *AvAttr, const swift::Decl *D) {
+    if (!isa<ValueDecl>(D))
+      return None;
+    
+    SmallVector<ValueDecl *, 4> lookupResults;
+    auto renamedParsedDeclName = swift::parseDeclName(AvAttr->Rename);
+    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
+    
+    auto declContext = D->getDeclContext();
+    Optional<ValueDecl *>renamedFuncDecl = None;
+
+    if (isa<ClassDecl>(D) || isa<ProtocolDecl>(D)) {
+      UnqualifiedLookup lookup(renamedDeclName.getBaseIdentifier(), declContext->getModuleScopeContext(), nullptr);
+      renamedFuncDecl = lookup.getSingleTypeResult();
+    } else {
+      declContext->lookupQualified(declContext->getSelfTypeInContext(),
+                                   renamedDeclName, NL_QualifiedDefault, NULL,
+                                   lookupResults);
+      for (auto candidate : lookupResults) {
+        if (!shouldInclude(candidate))
+          continue;
+        
+        if (candidate->getKind() != D->getKind())
+          continue;
+        
+        if (candidate->getKind() == DeclKind::Func && isa<FuncDecl>(candidate)
+            && cast<FuncDecl>(candidate)->getParameterLists().size() !=
+            cast<FuncDecl>(D)->getParameterLists().size())
+          continue;
+        
+        if (renamedFuncDecl)
+        /* TODO: Diagnose to compiler to tell the user that we found multiple candidates */;
```

Comment:
I think it's okay to just fail in this case, at least for now. Incremental improvements are good!

Reply:
How should I make it fail here? Just fallback to the name in the rename attribute?

Reply:
Yeah, that's what it's doing now. We could also consider not putting *anything* in the "rename" attribute when we can't get an ObjC name, and rely on the message. But for now, let's keep the existing behavior as our fallback.